crd/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::LabelSelector;
+use kube::CustomResource;
+use schemars::gen::SchemaGenerator;
+use schemars::schema::Schema;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::{from_value, json};
+use stackable_operator::Crd;
+use std::collections::HashMap;
+
+#[derive(Clone, CustomResource, Debug, Deserialize, JsonSchema, Serialize)]
+#[kube(
+    group = "nifi.stackable.tech",
+    version = "v1",
+    kind = "NiFiCluster",
+    shortname = "nifi",
+    namespaced
+)]
+#[kube(status = "NiFiStatus")]
+#[serde(rename_all = "camelCase")]
+pub struct NiFiSpec {
```

Comment:
Yeah i was inconsistent here. I used the OPA operator as template and for that i decided to use OpenPolicyAgent (without cluster). Here it would make sense, we do the same with (SparkCluster, ZookeeperCluster...). So i can change if we all agree.

Reply:
Yes. Sönke will change it in ZooKeeper as well so change NiFi to Nifi please in structs etc.